js/data/bucket.js

```diff
@@ -81,7 +81,7 @@ function Bucket(options) {
  * @private
  */
 Bucket.prototype.addFeatures = function() {
-    for (var i = 0; i < this.features.length; i++) {
+    for (var i = this.features.length - 1; i >= 0; i--) {
```

Comment:
I would like to note that this reversal might lead to some rendering bugs. While rendering order is reversed during translucent passes, it is in the expected order during opaque passes.